libs/vkd3d/state.c

```diff
@@ -6153,11 +6153,30 @@ HRESULT vkd3d_bindless_state_init(struct vkd3d_bindless_state *bindless_state,
     memset(bindless_state, 0, sizeof(*bindless_state));
     bindless_state->flags = vkd3d_bindless_state_get_bindless_flags(device);
 
+    if (!device_info->vulkan_1_2_features.shaderUniformTexelBufferArrayNonUniformIndexing ||
+            !device_info->vulkan_1_2_features.shaderStorageTexelBufferArrayNonUniformIndexing ||
+            !device_info->vulkan_1_2_features.shaderStorageBufferArrayNonUniformIndexing ||
+            !device_info->vulkan_1_2_features.shaderSampledImageArrayNonUniformIndexing ||
+            !device_info->vulkan_1_2_features.shaderStorageImageArrayNonUniformIndexing ||
+            !device_info->vulkan_1_2_features.descriptorBindingVariableDescriptorCount)
+    {
```

Comment:
Bit of a technicality, but any Vulkan implementation that advertizes `VkPhysicalDeviceVulkan12Features::descriptorIndexing` support must support all of the following:
```
shaderSampledImageArrayNonUniformIndexing
shaderStorageBufferArrayNonUniformIndexing
shaderUniformTexelBufferArrayNonUniformIndexing
```